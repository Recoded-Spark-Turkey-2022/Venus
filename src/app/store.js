// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
