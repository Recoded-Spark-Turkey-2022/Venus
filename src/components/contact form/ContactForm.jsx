import { useState } from 'react'
import Twitter from '../../assets/Twitter.svg'
import instagram from '../../assets/Instagram.svg'
import facebook from '../../assets/Facebook.svg'
import '../../index.css'

// import db(app) from firebase...........................

function ContactForm(){
    const [email, setEmail]=useState('');
    const [message, setMessage]=useState('');
  
    const handleSubmit = (e)=>{
    e.preventDefault();

    /* app.collection('contacts')
         .add({
            email:email,
            message:message
         })
         .then(()=>{
            alert('Your message has been submitted!');
         })
         .catch((error)=>{
            alert.(error.message)
         });

         setEmail('');
         setMessage('');
     */
        }
    return(
        <div className='w-full'>
            <form name="contact" className='flex flex-col justify-center md:justify-start mt-5 ' onSubmit={handleSubmit}>
            <div>{ /* eslint-disable-next-line jsx-a11y/label-has-associated-control */ }</div>
                <label htmlFor="email" className="text-darkGrey mb-2">
                    Email
                </label>
                <input  required id='email' value={email} onChange={(e)=>setEmail(e.target.value)} placeholder="Email" className="border-1 p-2 rounded-md border border-darkGrey bg-slate-50"/>
                <div>{ /* eslint-disable-next-line jsx-a11y/label-has-associated-control */ }</div>
                <label htmlFor="message" className="text-darkGrey mb-2 mt-2">
                    Message
                </label>
               <textarea value={message} onChange={(e)=>setMessage(e.target.value)} required id='message' className="border-1 p-2 rounded-md border border-darkGrey bg-slate-50 h-40" placeholder="Message"/>
                <div className='flex justify-center md:justify-start items-center'>
                <button type='submit' id='contact-button' className="w-32 rounded-full border border-mediumBlue mt-2 whitespace-nowrap py-2">
                    Send
                </button>
                </div>

            </form>
            <div className='flex justify-center md:justify-start'>
                <button type="button" className="mt-8"><img src={Twitter} alt="Twitter-logo" /></button>
                <button type="button" className="mt-8 pl-4"><img src={instagram} alt="Instagram-logo" /></button>
                <button type="button" className="mt-8 pl-4"><img src={facebook} alt="Facebook-logo" /></button>
            </div>
        </div>
    )


}
export default ContactForm;