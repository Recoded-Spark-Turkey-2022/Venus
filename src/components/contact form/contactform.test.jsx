import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import ContactForm from './ContactForm';

jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
      return {
        t: (str) => str
      };
    },
  }));

describe('Contact Form', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
            <Provider store={store}>
              <ContactForm />
            </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });