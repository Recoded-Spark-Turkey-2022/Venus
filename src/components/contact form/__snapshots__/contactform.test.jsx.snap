// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form renders correctly 1`] = `
<div
  className="w-full"
>
  <form
    className="flex flex-col justify-center md:justify-start mt-5 "
    name="contact"
    onSubmit={[Function]}
  >
    <label
      className="text-darkGrey mb-2"
      htmlFor="email"
    >
      Contact.email
    </label>
    <input
      className="border-1 p-2 rounded-md border m-0 border-darkGrey bg-slate-50"
      id="email"
      placeholder="Email"
      required={true}
      type="email"
    />
    <label
      className="text-darkGrey mb-2 mt-6"
      htmlFor="message-box"
    >
      Contact.message
    </label>
    <textarea
      className="border-1 p-2 outline-none rounded-md border max-h-[200px] border-darkGrey bg-slate-50 h-40"
      id="message"
      placeholder="Message"
      required={true}
    />
    <div
      className="flex justify-center md:justify-start items-center"
    >
      <button
        className="w-32 rounded-full mt-5 whitespace-nowrap py-2"
        id="mediumBlue-button"
        type="submit"
      >
        Contact.btn-text
      </button>
    </div>
  </form>
  <div
    className="flex justify-center md:justify-start"
  >
    <button
      className="mt-5"
      type="button"
    >
      <img
        alt="Twitter-logo"
        src="Twitter.svg"
      />
    </button>
    <button
      className="mt-5 pl-4"
      type="button"
    >
      <img
        alt="Instagram-logo"
        src="Instagram.svg"
      />
    </button>
    <button
      className="mt-5 pl-4"
      type="button"
    >
      <img
        alt="Facebook-logo"
        src="Facebook.svg"
      />
    </button>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
